numismatic/feeds/base.py

```diff
@@ -102,8 +105,9 @@ def subscribe(self, assets, currencies, channels, exchange=None,
         for symbol, channel in product(self._get_pairs(assets, currencies),
                                        channels):
             if self._websocket_client_class is not None:
-                websocket_client = self._websocket_client_class()
-                subscription = websocket_client.listen(symbol, channel)
+                if self.websocket_client is None:
+                    self.websocket_client = self._websocket_client_class()
+                subscription = self.websocket_client.listen(symbol, channel)
```

Comment:
This was causing multiple clients to be created and as a result multipe websockets. I think this needs to be overhauled because Luno will need to override this to handle their setup of multiple websockets.

Reply:
Yes exactly. That is what the code is doing now. I think that seems like the most sensible thing. A `WebsocketClient` is responsible for maintaining a single websocket connection and any subscriptions using that.